### ISSUE TITLE
Is this plugin still maintained? 

### ISSUE THREAD

**Issue by EYALIN** [admob-plus/admob-plus#547]
Is this plugin still maintained?

**Comment by kisimediaDE** [comment]
I have the same question. looking weekly for updates but nothing happens.

**Comment by EYALIN** [comment]
@ratson did some commits 10 days ago to a different repository, so we are lucky he is still using Github :-)

@ratson when you are back I will donate a monthly sponsorship.

**Comment by mujaffars** [comment]
Wating..

**Comment by mujaffars** [comment]
I observed that using plugin version> 1.11.1 (or we can say using the latest version 1.20.0 or 1.28.0) receives fewer ads. This means the variety of ads decreases and repetitive ads are shown. Does anyone face the same issue? @EYALIN @kisimediaDE ?

Using old versions like 1.11.1 or 1.10.0 works fine for Android os version <=10 but for Android os 11, 12 the app is crashing as soon as we try to open the app. 

Let me know if anyone observes the same.

**Comment by EYALIN** [comment]
@mujaffars after investigating, we didn't have such an issue

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
According to my knowledge this plug-in is using `Google-Mobile-Ads-SDK` version **8.13.0**

Google states that all version between **7** and **19** will [completely stop functioning](https://developers.google.com/admob/android/deprecation) from April 1st 2023.

Are my findings correct and what can we do - as a community - to update this plug-in and keep receiving ads?

![Clipboard 1](https://user-images.githubusercontent.com/15701574/227946769-3040ba11-587c-4b72-978d-b06bcaa0fe72.png)

**Comment by kisimediaDE** [comment]
I tried to fork the repo and run it locally because there are very good migration guides. 
But I was not able to run it 🫠 
(Have not much experience with Plugins like this)

So if someone else would give it a try and needs help with following the migration guides I'm more then open to help 🙂 

Here are so many people maybe we can find a group?

**Comment by EYALIN** [comment]
@kisimediaDE what I did is to fork it (you can take it from my repo/Pr) and  added some missing files->npm install
and then remove the admob plus package and reinstall it with the forked path for **Cordova folder**
please u[date us here so we will be able to help.

I upgraded the ADMOB SDK for IOS and Android and all works

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
> @kisimediaDE what I did is to fork it (you can take it from my repo/Pr) and added some missing files->npm install and then remove the admob plus package and reinstall it with the forked path for **Cordova folder** please update us here so we will be able to help.
> 
> I upgraded the ADMOB SDK for IOS and Android and all works

I did see (and did inspect the source-code) of your repo (some hours ago).

I did not implement it, yet (because I try to discover first what is going on).

In the beginning of the plug-in I did commit several changes as well (mainly non-coding issues), but after a year or so it became quit.

**But regarding your repo I do have two question, which I hope you can answer.**

1. In your commits I see only code-changes for the iOS version, not Android (besides some version bumps)? **Is this correct?**
2. Google says migrating from 7 - 19 to 20 (or higher) [will have many breaking](https://developers.google.com/admob/android/migration#migrate-to-v20) changes (causing the plug-in to stop). **Is this correct?**

![Clipboard 1](https://user-images.githubusercontent.com/15701574/228046547-1dcc104b-460e-4475-864b-3e1b6adf0f0c.png)

The plug-in in the `master` (here) still uses `Mobile Ads SDK - version 8.x`, while the newest version is already at `22` (so we are many version behind).

**And migrating from anything lower than `19` to a higher SDK, will have breaking changes... so the plug-in code (both Android and iOS) must have some thorough rewrites?**

I am a front-end / somehow back-end coder myself and somehow "good" in creating Cordova-apps, but only for Android and I don't have enough knowledge of the plug-in system.

**So let's see what we - the community - can do to maintain this plug-in.**

I am more than willing to support where and with whatever I can (but both Java and Swift are out of my scope).

**Comment by mjapps-games** [comment]
This is really bad news? We need to request/convince @ratson to do the plugin upgrade. He can help us with that, but I don't know how we can do that. He was not actively maintaining this plugin for long.

@EYALIN if you can do the upgrade and have the knowledge, please help us with that. Also please create npm package so that it can be used with volt.build for app creation.

**Comment by EYALIN** [comment]
@mjapps-games @1n3JgKl9pQ6cUMrW for Apple mobile ads SDK the latest is 10.3.0 (I don't know what version for Apple will be deprecated, can you please check as well?)

for Android, yes, it's going to be deprecated. but, we are talking about "PLAY_SERVICES_VERSION" for Android which is "20.6.0" in the current plugin ( I upgraded to 21.1.5)
I recommend to all, taking my pr locally, run npm install and install the package from the Cordova folder there.

**Comment by mjapps-games** [comment]
@EYALIN we are using https://volt.build/ for the creation of apk file, for that the package needs to be published on npm. Can you publish your package to https://www.npmjs.com/

Regarding your comment about "PLAY_SERVICES_VERSION" I am confused -- Are you saying current version will not stop showing ads in 1 week?

**Comment by mjapps-games** [comment]
@EYALIN Can you answer my first question?

we are using https://volt.build/ for the creation of apk file, and for that the package needs to be published on npm. Can you publish your package to https://www.npmjs.com/

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
Okay, there is a lot of confusing (because it is confusing), so let me try to bring some light (hopefully).

---

When I log in to my Admob account @ Google, I got a warning message about the deprecation of the `Google Mobile Ads SDK` (for Android).

The [latest version](https://developers.google.com/admob/android/sdk) for this SDK is `21.x` where our `Admob Cordova plug-in` stil uses `8x`.

![new](https://user-images.githubusercontent.com/15701574/228217108-ffa3b8ab-c9c9-45f4-bb6a-eee714158749.png)

---

From what I understand, this is **not** about the `Google Play Version` (SDK) but strictly the `Mobile Ads SDK` and that last one is already deprecated and will [totally stop functioning](https://developers.google.com/admob/android/deprecation) withing weeks.

![temp](https://user-images.githubusercontent.com/15701574/228216962-0b308fb4-c12c-4023-ade1-30c9299ed8c7.png)

Also this will impact **both** Android and iOS (please correct me, when I am wrong).

---

If we want to receive ads in Q2 of 2023, we must upgrade the `Google Mobile Ads SDK` from version 8 to at least 20 or even 21.

Doing so will [break](https://developers.google.com/admob/android/migration#migrate-to-v20) the existing plug-in, because there are so many version "in between".

![okay](https://user-images.githubusercontent.com/15701574/228217729-6989ef6d-b8f3-496b-84c5-2784ae6f3fc8.png)

---

**So from my understandings, we have about a couple of weeks and than the party is over.**

---

### **What we must do now;**

1. Confirm that we are talking about the `Google Mobile Ads SDK` and not the `Google Play services SDK`.
2. Find a solution for it,

Of course I can throw a couple of thousand bucks on the table and hire a Cordova / Android / Java developer myself, but that won't help the community.

---

I think **Ratson** does have it's own argument not to participate in this / his plug-in any longer, which we should all respect.

He did an amazing job building this from scratch, was fighting for the freedom (of speech) in his home country and did help us through the Covid-19 years... big should out to him.

_(and know we have to deal with this as our own problem)_

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
Again, it's confusing (all the version numbers, SDK, dependencies, etc...).

**So even when I am wrong, we must prepare for the future of this plug-in.**

I am willing to to pay for it, hire a developer, etc... but it would be nice if the community can handle this by itself.

There are so many Cordova Admob plug-ins out there, but all of them are years old and not trustworthy (besides this one).

**Comment by EYALIN** [comment]
@1n3JgKl9pQ6cUMrW for the Android it is the Google Play Services SDK.
I'm working on improving the patch and will try to upload it as an npm package, so it will be easy to consume it.

**Comment by mjapps-games** [comment]
Thank you @1n3JgKl9pQ6cUMrW for the brief details. We need to plan how we are going to upgrade it. I don't have much experience on Java end, but reviewed some code and tried to understand it. It is a little bit complicated to understand for someone who doesn't know each detail about it. If someone can provide a tutorial about it then it will be helpful.

We can also think about docker container creation, which can be used for the testing purposes -- Just a thought.

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
> Thank you @1n3JgKl9pQ6cUMrW for the brief details. We need to plan how we are going to upgrade it. I don't have much experience on Java end, but reviewed some code and tried to understand it.

I started with Java way back in the '90 but lot's have changed since than.

For `Android` we need to maintain the code in `Java` but the `iOS` version needs `Swift` (as far as I know).

For now we are good with the newest update from @EYALIN but withing one or two years, this plug-in really needs to get / receive new updates.

Luckily the sourcecode is here to stay and quite easy to read / maintain (for those with enough knowledge).

@ratson really did an outstanding job.

**Comment by jasonandress** [comment]
Hey everyone, sorry for the simple seeming question. How do I install the plugin from @EYALIN GitHub? @1n3JgKl9pQ6cUMrW

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
> Hey everyone, sorry for the simple seeming question. How do I install the plugin from @EYALIN GitHub? @1n3JgKl9pQ6cUMrW

`CLI` with repo `https://github.com/EYALIN/admob-plus.git`

Or download the .zip at `https://github.com/EYALIN/admob-plus/archive/refs/heads/master.zip` and install it manually (on your local machine) by referring to the path you unzipped it.

Basically the [install process](https://admob-plus.github.io/docs/cordova/installation) is the same as the original plug-in, but only with a different endpoint.

**Comment by EYALIN** [comment]
> ## Situation
> Okay, things are getting clearer now and I will try **to sort things out** (_for myself and everybody else_).
> 
> 1. This plug-in is awesome (for many of us), but needs **upgrades** and **maintenance**.
> 2. Since the original creator is somehow **AFK**, we (the community) must / can / will do this **ourself**.
> 
> ## Plug-in versions
> * The most current version (made by **@ratson** ( of this plug-in is `1.28.0` and can be found [here](https://github.com/admob-plus/admob-plus).
> * But because dependencies are running out of time (more on that below) user **@EYALIN** updated both the most important SDK's for both `iOS` and `Android`, bumping the version to `1.29.0.` which can be found [here](https://github.com/EYALIN/admob-plus).
> 
> Like I mentioned above, the plug-in remains the same but the dependencies are updates (preventing ads to disappear and / or other misbehavior).
> 
> ## SKD overview
> This plug-in uses the `Google Mobile Ads SDK` and does this both for `Android` and `iOS`.
> 
> The `iOS` version can be seen as a "native" SDK, so it's stand-alone and must be integrated "as is" in your app (since Google can't provide their complete framework insides Apples software environment).
> 
> The `Android` version also uses this `Google Mobile Ads SDK` but here it's part of the complete "Android" framework, called `Google Play services`.
> 
> **And both SDK do need an update soon.**
> 
> For `Android` the deprecation schedule of the `Mobile Ads SDK` can be found [here](https://developers.google.com/admob/android/deprecation) where all the version numbers refer to the `Google Play services`.
> 
> And for `iOS` the complete overview of "needs to be updated" versions of the same `SDK` can be found [here](https://developers.google.com/admob/ios/migration)
> 
> ## Used SDK's
> The original plug-in uses the `Mobile Ads SDK` for `Android` (which is part of the `Google Play services`) version number `20.6.0` (plug-in version `1.28.0`) and is updated in `1.29.0` to version `21.5.0`.
> 
> For the `iOS` version, the `SDK` is updated from `8.13.0` to `10.1.0`.
> 
> All those mentioned changes can be tracked in the filediff [here](https://github.com/admob-plus/admob-plus/pull/564/files)
> 
> ## Deprecation schedule
> Google also announced a deprecation / sunset schedule for the mentioned version numbers.
> 
> Once again, check this out for `Android` [here](https://developers.google.com/admob/android/deprecation) and `iOS` [overhere](https://developers.google.com/admob/ios/migration)
> 
> For `Android` the active version `20.6.0` is already deprecated since `Q1 2023` and will completely stop functioning in `Q2 2024`.
> 
> So when you do **not upgrade** from plug-in version `1.28.0` to `1.29.0` it will stop working within a year.
> 
> The newest version in `1.29.0.` is now set to be `21.5.0` which will be deprecated in `Q1 2024` and stop functioning in `Q2 2025`.
> 
> For `iOS` the current version is set to `8.13.0` where the newest version is `10.1.0`.
> 
> This update will not break anything within your app but on both `iOS 10` and `iOS 11` the ads will stop showing (using `SDK 10.1.0`), according to Apple / Google.
> 
> ### Sidenote Android
> All `Android` users of the new version `1.29.0` are "safe" until `Q2 2025` but it's clear that we must maintain the version numbers year by year.
> 
> The same is for `iOS` but since that ecosystem is not part of Google, it uses different rules, schedules and sunsets.

@1n3JgKl9pQ6cUMrW  thank you very much for the detailed explanation! highly appreciated

**Comment by jasonandress** [comment]
It's a plain cordova. I'm not asking how to install a plugin, I can do that. What's specific about this repo is there are many plugins.

I've tried this 
cordova plugin add https://github.com/EYALIN/admob-plus/tree/master/packages/cordova —variable ...

I've tried the same but from my local, neither works.

Can you help?

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
**Maybe it's better to open a new topic for this, so others can also find answers for the same problem.**

---

This repo has just one plug-in, but it's wrapped in difference namespaces for different frameworks.

Ionic, Cordova (native), React, etc...

Your example is not a valid .git endpoint, it's the main repo.

Just [download](https://github.com/EYALIN/admob-plus/archive/refs/heads/master.zip) the package and unzip it to a local folder.

Than install it by pointing to the Cordova folder you need.

`cordova plugin add /random/folder/admob-plus-master/packages/cordova/`

**Comment by ollm** [comment]
Before installing the repository, you have to build the js files  (Change the repo and package to the one you use), the `npm install` will probably give some errors, ignore them.

```
git clone https://github.com/admob-plus/admob-plus.git
cd admob-plus
npm install
cd packages/cordova
npm install tslib@2.5.0 --save-dev
npm install
npm run prepublishOnly
```
Now you can install the repo as follows.

`cordova plugin add /you/folder/name/admob-plus/packages/cordova --save --variable ...`

This only needs to be done again if you modify any of the js in the repo.

**Comment by mjapps-games** [comment]
@ollm can this updated package be available on npm (https://www.npmjs.com)? So it will be easy to use via https://volt.build/ service?

**Comment by ollm** [comment]
> @ollm can this updated package be available on npm (https://www.npmjs.com)? So it will be easy to use via https://volt.build/ service?

You can use my `admob-plus-cordova`  fork with [WebView API for Ads](https://github.com/admob-plus/admob-plus/pull/539), which also has the @EYALIN changes with the updated SDK.

https://www.npmjs.com/package/admob-plus-cordova-webview-api-for-ads

**Comment by WynOwen72** [comment]
Newbie question as I'm now confused. My app is Ionic\Capacitor\Vue3 (for Android) so want should I use?

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
Maybe it's better to open a new topic for this, so others can also find answers for the same problem.

---

I think the way to install the new plug-in is described way too difficult (correct me if I am wrong).

I did not have the time to install it myself, since I have to test is for weeks first, but it's not that difficult (correct me if I am wrong).

Just download the newest repo, install the Cordova version of the plugin and you're done.

---

Here you'll see all the options how to do it;

https://doc.xuwenliang.com/docs/cordova/1338

Basically it's `cordova plugin add /path/to/**directory**`.

Where `directory` should contain the `plugin.xml`.

---

Like I said; I was not able to test it myself, but 9 out of 10 it's just like this (but maybe the new version is different... please tell us here).

**Comment by ollm** [comment]
> Like I said; I was not able to test it myself, but 9 out of 10 it's just like this (but maybe the new version is different... please tell us here).

@1n3JgKl9pQ6cUMrW This is correct in most plugins, but `admob-plus` does not have in repo the necessary js files that are generated when building, for this reason, the steps that I mention above are necessary so that the js files are generated in the local repository, unless the js files are push to the repository.

**Comment by ollm** [comment]
> In that case, this should be done? (please correct when wrong)

Yes, is necessary to generate the js files in `admob-plus/packages/cordova/www` by building the TypeScript files that are in `admob-plus/packages/cordova/src/www`, since the plugin will use these files once installed.

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
For those who don't have .git (for whatever reason) installed globally on the system, you can also run the commands from a portable version of .git (which is handy also for other reasons, by the way).

https://github.com/sheabunge/GitPortable/releases/tag/v2.27.0-devtest.1

**Comment by DrMinh** [comment]
```shell
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: example.admob.basic@0.16.15
```

**Comment by jasonandress** [comment]
@ollm - thank you that worked!

@DrMinh I got some errors when using an old version of npm v12. Everything worked when i switched to npm v16

**Comment by DrMinh** [comment]
> @ollm - thank you that worked!
> 
> @DrMinh I got some errors when using an old version of npm v12. Everything worked when i switched to npm v16

what did you mean v16? I checked lastest version of npm is v9.6.3?

**Comment by WynOwen72** [comment]
> what did you mean v16? I checked lastest version of npm is v9.6.3?

I guess they mean Nodejs version.

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
- `Node` latest version `18.x`
- `NPM` latest version `9.x`

@DrMinh can you please elaborate what errors you received and what you did to solve them?

And - of course - is the newest plug-in working for you?

(and which repo did you use for this).

Let's help each other (out), here 👍

**Comment by DrMinh** [comment]
I already use this plugin for my project and I try to change to EYALIN version. I follow your instruction and got errors
```powershell
git clone https://github.com/EYALIN/admob-plus.git
cd admob-plus
npm install <--- I got error in this step
cd packages/cordova
npm install tslib@2.5.0 --save-dev
npm install
npm run prepublishOnly
``` 
Error:
```shell
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: example.admob.basic@0.16.15
npm ERR! Found: appium@1.22.3
npm ERR! node_modules/appium
npm ERR!   dev appium@"^1.22.3" from example.admob.basic@0.16.15
npm ERR!   examples/cordova
npm ERR!     example.admob.basic@0.16.15
npm ERR!     node_modules/example.admob.basic
npm ERR!       workspace examples\cordova from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer appium@"^2.0.0-beta.40" from appium-android-driver@5.8.12
npm ERR! node_modules/appium-android-driver
npm ERR!   dev appium-android-driver@"^5.2.7" from example.admob.basic@0.16.15
npm ERR!   examples/cordova
npm ERR!     example.admob.basic@0.16.15
npm ERR!     node_modules/example.admob.basic
npm ERR!       workspace examples\cordova from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by EMI-INDO** [comment]
I will rewrite everything, according to my own needs ( Cordova Android platform )
https://github.com/EMI-INDO/emi-indo-cordova-plugin-admob

**Comment by 1n3JgKl9pQ6cUMrW** [comment]
> I will rewrite everything, according to my own needs ( Cordova Android platform ) https://github.com/EMI-INDO/emi-indo-cordova-plugin-admob

**Interesting.**

Are you rewriting it because the original isn't maintained any longer?

And is yours already ready to be used (e.g. is it serving ads and does one need to compile the Typescript assets by themself)?

**Comment by EMI-INDO** [comment]
@1n3JgKl9pQ6cUMrW 

### **Yes now I added a new feature Mediation, It works fine.**

<img width="1411" alt="2023-04-03_061951" src="https://user-images.githubusercontent.com/78555833/229385187-0bfc076b-bdbe-457e-ba35-51dbee893e35.png">

**Comment by mjapps-games** [comment]
@EMI-INDO Thank you for your initiative

Can you assure us that you will support this plugin (emi-indo-cordova-plugin-admob) for long. So that we can switch to this plugin.

**Comment by EYALIN** [comment]
@1n3JgKl9pQ6cUMrW  @mjapps-games @EMI-INDO @DrMinh @WynOwen72 @ollm @jasonandress @kisimediaDE @mujaffars 
 I duplicated the plugin here:https://www.npmjs.com/package/community-admob-plus-cordova

**Now you can install it with the npm**
for now, it's for Cordova only, if someone wants for different one, please update here.

will test it in the next few days, but it looks like it works.

**Comment by mjapps-games** [comment]
Thank you @EYALIN 
Can you let us know which should be used from below (specifications)

```
admob-plus-cordova-webview-api-for-ads
community-admob-plus-cordova
```

**Comment by EYALIN** [comment]
> Thank you @EYALIN Can you let us know which should be used from below (specifications)
> 
> ```
> admob-plus-cordova-webview-api-for-ads
> community-admob-plus-cordova
> ```

community-admob-plus-cordova

**Comment by EMI-INDO** [comment]
**SDK Migrate from v21 to v22** 
https://github.com/EMI-INDO/emi-indo-cordova-plugin-admob/releases/tag/%400.0.4

**Comment by A32rj3gebebr** [comment]
> **SDK Migrate from v21 to v22** https://github.com/EMI-INDO/emi-indo-cordova-plugin-admob/releases/tag/%400.0.4

Will there also be a current version for iOS?

**Comment by EYALIN** [comment]
@A32rj3gebebr This new plugin  "community-admob-plus-cordova" (https://github.com/EYALIN/community-admob-plus) has the most up to date Ads SDK for IOS. also, we changed all the needed functions in order to support it.
tested for 2 months now.

**Comment by ollm** [comment]
@EYALIN you want me to push the changes from my fork to yours? My fork includes support for [WebView API for Ads](https://github.com/admob-plus/admob-plus/pull/539)  and a fix for the following bug on iOS https://github.com/admob-plus/admob-plus/issues/530 (https://github.com/admob-plus/admob-plus/pull/539/commits/cf773d9cef81c7b111ab45542a841f95c70e595b)

**Comment by DrMinh** [comment]
> @A32rj3gebebr This new plugin "community-admob-plus-cordova" (https://github.com/EYALIN/community-admob-plus) has the most up to date Ads SDK for IOS. also, we changed all the needed functions in order to support it. tested for 2 months now.

cool! Good to know the plugin still active.

**Comment by EYALIN** [comment]
> @EYALIN you want me to push the changes from my fork to yours? My fork includes support for [WebView API for Ads](https://github.com/admob-plus/admob-plus/pull/539) and a fix for the following bug on iOS #530 ([cf773d9](https://github.com/admob-plus/admob-plus/commit/cf773d9cef81c7b111ab45542a841f95c70e595b))

that will be great, thanks! :-)
also, I tried it in the past,but couldn't use it.
after you will push your fork, I will try to test it again.

אד
ארא

**Comment by ollm** [comment]
I sent my pull request https://github.com/EYALIN/community-admob-plus/pull/1 

When you can, also activate the issues tab in you fork

**Comment by Amaeta** [comment]
> > Thank you @EYALIN Can you let us know which should be used from below (specifications)
> > ```
> > admob-plus-cordova-webview-api-for-ads
> > community-admob-plus-cordova
> > ```
> 
> community-admob-plus-cordova

I tried installing through Git, I get this error:
Installing "community-admob-plus-cordova" for android
Error during processing of action! Attempting to revert...
Failed to install 'community-admob-plus-cordova': CordovaError: Uh oh!

I

**Comment by EMI-INDO** [comment]
- New Features | New Plugin |  Bug fix

https://github.com/EMI-INDO/emi-indo-cordova-plugin-admob/releases/tag/v0.0.5

**Comment by Amaeta** [comment]
> Test the app in the Android Studio Emulator so you can see the errors in logcat.

>>> loadUrl(javascript:cordova.plugins.backgroundMode._setActive(true);cordova.plugins.backgroundMode.on('activate', null);cordova.plugins.backgroundMode.fireEvent('activate',null);)
2023-04-11 19:17:59.034 13047-13047 CordovaActivity         com.ik..                 D  Resumed the activity.
2023-04-11 19:17:59.035 13047-13047 CordovaWebViewImpl      com.ik..                 D  >>> loadUrl(javascript:cordova.plugins.backgroundMode._setActive(false);cordova.plugins.backgroundMode.on('deactivate', null);cordova.plugins.backgroundMode.fireEvent('deactivate',null);)
2023-04-11 19:17:59.198 13047-13047 AndroidRuntime          com.ik..                 D  Shutting down VM
2023-04-11 19:17:59.292 13047-13047 AndroidRuntime          com.ik..                 E  FATAL EXCEPTION: main
                                                                                                    Process: com.ik.., PID: 13047
                                                                                                    java.lang.RuntimeException: Unable to create service de.appplant.cordova.plugin.background.ForegroundService: java.lang.IllegalArgumentException: com.ik..: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
                                                                                                    Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
                                                                                                    	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4498)
                                                                                                    	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2161)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:201)
                                                                                                    	at android.os.Looper.loop(Looper.java:288)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7872)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
                                                                                                    Caused by: java.lang.IllegalArgumentException: com.ik..: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
                                                                                                    Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
                                                                                                    	at android.app.PendingIntent.checkFlags(PendingIntent.java:401)
                                                                                                    	at android.app.PendingIntent.getActivityAsUser(PendingIntent.java:484)
                                                                                                    	at android.app.PendingIntent.getActivity(PendingIntent.java:470)
                                                                                                    	at android.app.PendingIntent.getActivity(PendingIntent.java:434)
                                                                                                    	at de.appplant.cordova.plugin.background.ForegroundService.makeNotification(ForegroundService.java:223)
                                                                                                    	at de.appplant.cordova.plugin.background.ForegroundService.makeNotification(ForegroundService.java:163)
                                                                                                    	at de.appplant.cordova.plugin.background.ForegroundService.keepAwake(ForegroundService.java:132)
                                                                                                    	at de.appplant.cordova.plugin.background.ForegroundService.onCreate(ForegroundService.java:100)
                                                                                                    	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4485)
                                                                                                    	... 9 more
2023-04-11 19:17:59.436 13047-13047 Process                 com.ik..                 I  Sending signal. PID: 13047 SIG: 9
---------------------------- PROCESS ENDED (13047) for package com.ik.. ----------------------------
2023-04-11 19:18:00.616   561-1861  AppOps                  pid-561                              E  Bad call made by uid 1000. Package "com.ik.." does not belong to uid -1.
2023-04-11 19:18:00.617   561-1861  AppOps                  pid-561                              E  noteOperation
                                                                                                    java.lang.SecurityException: Specified package "com.ik.." under uid -1 but it is not
                                                                                                    	at com.android.server.appop.AppOpsService.verifyAndGetBypass(AppOpsService.java:4694)
                                                                                                    	at com.android.server.appop.AppOpsService.noteOperationUnchecked(AppOpsService.java:3453)
                                                                                                    	at com.android.server.appop.AppOpsService.noteOperationImpl(AppOpsService.java:3441)
                                                                                                    	at com.android.server.appop.AppOpsService.-$$Nest$mnoteOperationImpl(Unknown Source:0)
                                                                                                    	at com.android.server.appop.AppOpsService$CheckOpsDelegateDispatcher.lambda$noteOperation$4(AppOpsService.java:7640)
                                                                                                    	at com.android.server.appop.AppOpsService$CheckOpsDelegateDispatcher.$r8$lambda$oOwUoG55BBmJ6yiwhp3OAq3tQ30(Unknown Source:0)
                                                                                                    	at com.android.server.appop.AppOpsService$CheckOpsDelegateDispatcher$$ExternalSyntheticLambda1.apply(Unknown Source:32)
                                                                                                    	at com.android.server.policy.AppOpsPolicy.noteOperation(AppOpsPolicy.java:227)
                                                                                                    	at com.android.server.appop.AppOpsService$CheckOpsDelegateDispatcher.noteOperation(AppOpsService.java:7638)
                                                                                                    	at com.android.server.appop.AppOpsService.noteOperation(AppOpsService.java:3425)
                                                                                                    	at com.android.internal.app.IAppOpsService$Stub.onTransact(IAppOpsService.java:479)
                                                                                                    	at android.os.Binder.execTransactInternal(Binder.java:1280)
                                                                                                    	at android.os.Binder.execTransact(Binder.java:1244)

**Comment by EYALIN** [comment]
@Amaeta are the errors related to which plugin? what version?

**Comment by ollm** [comment]
@Amaeta the crash in your app is caused by the plugin `cordova-plugin-background-mode`.

You can try removing the plugin.

``` bash
cordova plugin remove cordova-plugin-background-mode
```
Or apply one of the solutions discussed here https://github.com/katzer/cordova-plugin-background-mode/issues/577

**Comment by Amaeta** [comment]
> @Amaeta the crash in your app is caused by the plugin `cordova-plugin-background-mode`.
> 
> You can try removing the plugin.
> 
> ```shell
> cordova plugin remove cordova-plugin-background-mode
> ```
> 
> Or apply one of the solutions discussed here [katzer/cordova-plugin-background-mode#577](https://github.com/katzer/cordova-plugin-background-mode/issues/577)

Thanks a bunch. Removed the background-mode plugin and it is working now. The only issue is that I can't get the native ad to show. Interstatia and banner works well. Could be my code?

function showNativeAd(){
    
        setTimeout(function() {
		ad.show({
          x: 0,
          y: 30,
          width: window.screen.width,
          height: 300,
        });
		}, 5000);
      
         setTimeout(function() {
              ad.hide();
            }, 5000);
      ad.showWith(document.getElementById('nativeads')); //id of the div I want to display my native ad. Using Cordova
  };

**Comment by EMI-INDO** [comment]
@Amaeta 

Ads must be loaded before displaying

```
load()
      .then(() =>
        ad.show({.....}) 

// ad.hide() 

```
https://github.com/admob-plus/admob-plus/blob/master/examples/cordova/www/js/index.js

**Comment by Amaeta** [comment]
> @Amaeta
> 
> Ads must be loaded before displaying
> 
> ```
> ad.load()
>       .then(() =>
>         ad.show({.....}) 
> 
> // ad.hide() 
> ```
> 
> https://github.com/admob-plus/admob-plus/blob/master/examples/cordova/www/js/index.js

Yes, I have  ad.load(); first. Do I need to install the native admob plugin differently or does community-admob-plus plugin enough?

**Comment by ollm** [comment]
> > @Amaeta
> > Ads must be loaded before displaying
> > ```
> > ad.load()
> >       .then(() =>
> >         ad.show({.....}) 
> > 
> > // ad.hide() 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > https://github.com/admob-plus/admob-plus/blob/master/examples/cordova/www/js/index.js
> 
> Yes, I have ad.load(); first. Do I need to install the native admob plugin differently or does community-admob-plus plugin enough?

Yes, to use native ads it is necessary to install the `admob-plus-cordova-native` plugin or a custom fork of it.

You can get more information in the native ad documentation: https://admob-plus.github.io/docs/cordova/ads/native

**Comment by EMI-INDO** [comment]
###  emi-indo-cordova-plugin-admob ( New Platform IOS )

https://github.com/EMI-INDO/emi-indo-cordova-plugin-admob/discussions/3

**Comment by EMI-INDO** [comment]
## New Features AppOpenAd ( support Any mediation )

https://github.com/EMI-INDO/emi-indo-cordova-plugin-admob

**Comment by degroundshaker** [comment]
> @1n3JgKl9pQ6cUMrW @mjapps-games @EMI-INDO @DrMinh @WynOwen72 @ollm @jasonandress @kisimediaDE @mujaffars I duplicated the plugin here:https://www.npmjs.com/package/community-admob-plus-cordova
> 
> **Now you can install it with the npm** for now, it's for Cordova only, if someone wants for different one, please update here.
> 
> will test it in the next few days, but it looks like it works.

I have installed it from npm, but i do not understand how to use it in my ionic app.

**Comment by EYALIN** [comment]
@degroundshaker can you be more specific? and you can open  a new issue in the community-admob-plus-cordova plugin repo

**Comment by EMI-INDO** [comment]
## Update New Features 
https://github.com/EMI-INDO/emi-indo-cordova-plugin-admob/releases/tag/v.1.0.9

**Comment by EYALIN** [comment]
I see that @ratson is back and merged some code. @ratson  are you back? we are all waiting for you and highly appreciate your work

**Comment by EYALIN** [comment]
@ratson please update us if you are back.
personally, I want to give your sponsorship, and I'm sure much more people here

**Comment by ratson** [comment]
@EYALIN See my plan in https://github.com/admob-plus/admob-plus/pull/539#issuecomment-1556069619

For those need to  the latest SDKs, please try the latest `2.0.0-alpha.1` version, it should be API compatible, except require Kotlin support.

@EMI-INDO  Good work! For those looking for an alternative, feel free to try it out.
